### ISSUE TITLE
Meta: Can we automate checking for duplicate stories in past issues?

### ISSUE THREAD

**Issue by john-light** [aragon/aragon-monthly#91]
It's possible that someone might try to share something that has been featured in a previous issue. While it is not the end of the world if a duplicate story is added, it would be nice to reduce such occurrences. The Editors might miss duplicates, since after so many issues it might be easy to forget or not know what has been covered before.

We should make a bot that can scan new Issues and Pull Requests and detect when a link has been submitted to previous issues of the newspaper.

If there is more interest in creating a bot for this purpose then we can create a bounty that pays whoever makes the bot that does this. Let me know what you think!

**Comment by leolower** [comment]
Hi @john-light if you are still interested I'd love to help with this issue.

**Comment by john-light** [comment]
@leolower yes please! We're starting to get a lot of submissions and I've noticed a duplicate at least once, but it doesn't mean we'll catch it every time. What do you have in mind for a solution?

**Comment by leolower** [comment]
Cool! 
I was thinking on a [probot](https://github.com/probot/probot) app hosted on [Now](https://zeit.co/now) or Heroku. 
It would listen for new issues / PRs, extract the links from the text and compare them against other issues using Github's API or a separate DB. If a duplicate is found it could post a comment or add a label to the issue. 
It can probably be adapted to the specific workflow you already have. What do you think?

**Comment by john-light** [comment]
That sounds great. Would be good to have it give a link to the issue with the duplicate link in the comment so the submitter can see if the article was previously accepted or not.

**Comment by status-open-bounty** [comment]
![Contract deploying](https://openbounty.status.im/img/deploying_contract.png)
Transaction: [0x3eaea975ad09308f4d2d2598e26e97696f8f4d878d6986932b97443125ee3591](https://etherscan.io/tx/0x3eaea975ad09308f4d2d2598e26e97696f8f4d878d6986932b97443125ee3591)

**Comment by john-light** [comment]
@leolower I just added a bounty to this PR. You'll need to create a Status Open Bounty account and submit a PR to this repo that closes this issue to claim the bounty. If a PR is unnecessary to claim the bounty then we should discuss with @Smokyish the best way to do this -- either manually send you the bounty out of band or submit a PR adding some documentation about the bot would probably suffice.

**Comment by leolower** [comment]
Yes, I would recommend keeping the bot on a separate repository and adding instructions for submitters on this repo's readme of issue templates.

**Comment by leolower** [comment]
Hello, we've made some progress on the bot.

I have configured it on a dummy repository as an example: https://github.com/leolower/drone-test/issues 

On that repo you can see that issues have different links and whenever a new issue is posted the bot checks all new links and if a duplicate is found, it posts a comment with links to the original issue.

Let me know what you think!

**Comment by john-light** [comment]
Very cool, thanks @leolower! I will take a look. Also pinging @Smokyish to take a look.

**Comment by leolower** [comment]
By the way, you can submit issues with links on that repo to test it. If you like it I can provide instructions on how to set it up on your repo and if you prefer, run it on a server of your own.

The code is here: https://github.com/protofire/aragon-links-bot

**Comment by leolower** [comment]
Hi! Any news on this? Let me know if you have suggestions or changes you want to make on the app.
